packages/data_structures/src/byte_reader.cairo

```diff
@@ -562,19 +576,10 @@ impl ByteReaderLenImpl<T, +Len<T>> of Len<ByteReaderState<T>> {
     }
 }
 
-impl ByteArrayIndexViewAsSnapshotImpl of IndexView<ByteArray, usize, @u8> {
-    #[inline(always)]
-    fn index(self: @ByteArray, index: usize) -> @u8 {
-        @self.at(index).expect('Index out of bounds')
-    }
-}
-
-impl ByteReaderIndexViewImpl<
-    T, impl TIndexView: IndexView<T, usize, @u8>
-> of IndexView<ByteReaderState<T>, usize, @u8> {
+impl IntoU8Impl of Into<u8, @u8> {
```

Comment:
what am i watching here ? is this needed for some trait ?

Reply:
this is the one thing I'm a bit puzzled, and had a lot pf back and forth with Ariel on Discord.
This is the only way I found to have this file working.
If you'd be kind enough to look into it, that'd be nice...